File: 프로그래머스/lv1/12901. 2016년/2016년.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(int a, int b) {
    int month[12] = {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
    int sum= 0;
    char* answer = (char*)calloc(sizeof(char),4);
    for(int i=0; i<a-1; i++){
        sum += month[i];
    }
    sum += b;
    switch(sum % 7){
        case 0:
        answer = "THU";
        break;
            
        case 1:
        answer = "FRI";
        break;
            
        case 2:
        answer = "SAT";
        break;
            
        case 3:
        answer = "SUN";
        break;
            
        case 4:
        answer = "MON";
        break;
            
        case 5:
        answer = "TUE";
        break;
            
        case 6:
        answer = "WED";
        break;      
    }
    return answer;
}

File: 프로그래머스/lv1/12901. 2016년/2016년.cs
public class Solution {
    public string solution(int a, int b) {
    string answer = "";
        string[] day = new string[] { "FRI", "SAT", "SUN", "MON", "TUE", "WED", "THU"};
        int[] days = new int[] {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        
        int totalDays = 0;
        for(int i=0; i<a-1; i++)
        {
            totalDays += days[i];
        }
        totalDays += b -1;
        
        answer = day[totalDays % 7];
        return answer;
    }
}